src/workerd/api/crypto/impl.c++

```diff
@@ -298,4 +298,28 @@ bool CSPRNG(kj::ArrayPtr<kj::byte> buffer) {
 
   return false;
 }
+
+
+kj::Maybe<kj::ArrayPtr<const kj::byte>> tryGetAsn1Sequence(kj::ArrayPtr<const kj::byte> data) {
+  if (data.size() < 2 || data[0] != 0x30)
+    return kj::none;
+
+  if (data[1] & 0x80) {
+    // Long form.
+    size_t n_bytes = data[1] & ~0x80;
+    if (n_bytes + 2 > data.size() || n_bytes > sizeof(size_t))
+      return kj::none;
+    size_t length = 0;
+    for (size_t i = 0; i < n_bytes; i++)
+      length = (length << 8) | data[i + 2];
+    auto start = 2 + n_bytes;
+    auto end = start + kj::min(data.size() - 2 - n_bytes, length);
```

Comment:
By doing kj::min, this ends up returning truncated data if the input data is shorter than it should be as indicated by the length field instead of rejecting it. Is that what we want to do here?

Reply:
This would match the behavior in node.js I believe, but I will verify for sure when I start implementing the tests. So if this needs to change it can be changed then.